parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -110,8 +119,94 @@ public boolean isSmallerThan(long size) {
    */
   @Deprecated
   public void updateStats(Binary min_value, Binary max_value) {
-    if (comparator().compare(min, min_value) > 0) { min = min_value.copy(); }
-    if (comparator().compare(max, max_value) < 0) { max = max_value.copy(); }
+    if (comparator().compare(min, min_value) > 0) { min = truncator.truncateMin(min_value.copy(), truncateLength); }
```

Comment:
This method is invoked for every values to be written to the file. This is a huge overhead and useless. I would do the truncate just before writing to the file only once for each column chunk.

Reply:
Agree! Removed it.